pkg/connector/mock/builder.go

```diff
@@ -44,6 +44,8 @@ func (b Builder) Init(c connector.Connector, id string, cfg connector.Config) er
 		m.EXPECT().Type().Return(connector.TypeSource).AnyTimes()
 		m.EXPECT().ID().Return(id).AnyTimes()
 		m.EXPECT().Config().Return(cfg).AnyTimes()
+		m.EXPECT().SetProvisionedBy(connector.ProvisionTypeAPI).AnyTimes()
```

Comment:
We shouldn't expect this method to be called, this can and should be set up in the tests that need it (same for destination). This comment may be redundant if we remove the method entirely.